cornell-dti/course-reviews-react-2.0
# Meteor Shower: Washing Away Meteor

### Meteor Shower: Washing Away Meteor :shower: 
> That's one small step for an app, one giant leap for appkind :man_astronaut: 

This pull request works to remove Meteor, in its entirety, from CU Reviews. It does so by supplying basic shims for Meteor's method and subscription APIs and migrating database calls to `mongoose`.

The client is moved to `create-react-app` in `client/` whilst the server uses express and Node.js in `server/`.

- [X] Convert client to `create-react-app`
- [X] Shim basic Meteor behavior
- [X] Use Yarn workspaces as a replacement mechanism of code sharing between frontend and backend

This is based on Sam's TypeScript work :heart: 

### Test Plan

To run the app:

1. check out the branch.
2. Install [yarn](https://classic.yarnpkg.com/lang/en/)
3. `yarn install` to install all dependencies
4. Put `MONGODB_URL=[SECRET_URL]` in `server/.env`
5. Start backend by `yarn workspace server start`
6. Start frontend by `yarn workspace client start`
7. Play with it on localhost:3000

Aggressively test for regressions against cureviews.org

### Notes

Once the `blue-redesign` goes into master, rebase and change the base to `master` on github.

Once this diff is done, we can close #118, close #123, close #127, close #135, since it contains all the fixes in those PRs.
